spyder/plugins/editor/lsp/client.py

```diff
@@ -89,6 +85,13 @@ def __init__(self, parent,
         self.transport_args = [sys.executable, '-u',
                                osp.join(LOCATION, 'transport', 'main.py')]
         self.external_server = server_settings.get('external', False)
+        self.stdio = server_settings.get('stdio', False)
+        # Setting stdio on implies that external_server is off
+        if self.stdio and self.external_server:
+            error = ('If server is set to use stdio communication, '
+                     'then it cannot be an external server')
```

Comment:
Why is this restriction necessary?

Reply:
Unless an user gives a process ID to redirect its stdin and stdout pipes, I think is impossible to connect to an external server using stdio